lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -372,13 +303,11 @@ struct ConvertLayoutOpUsingLinearLayoutsConversion
       return transferWithinLane(*c, op, adaptor, rewriter);
     }
 
-    if (std::optional<LinearLayout> c = conversion.divideRight(
-            LinearLayout::identity1D(numBlocks, kBlock, kBlock));
-        c.has_value()) {
-      return transferWithinBlock(*c, op, adaptor, rewriter);
-    }
-
-    return transferWithinBlockGroup(conversion, op, adaptor, rewriter);
+    // TODO(Keren): LinearLayout doesn't support cross-CTA conversion yet.
+    if (isCrossCTAConversion(conversion))
```

Comment:
I'd move this check into transferWithinBlockOrGroup.  Otherwise it's confusing, the function is called "transferWithinBlockOrGroup" but we refuse to call it when we're doing a transfer within a group?